docs/user/tutorial.rst

```diff
@@ -1304,10 +1304,201 @@ Inspecting the application now returns:
         ⇒ /images/{name} - Item:
            └── GET - on_get
 
-.. Query Strings
-.. -------------
+Query Strings
+-------------
+Now that we are able to get the images from the service, we need a way to get a list available images. We have already set up this route. Before testing this route let's change its output format back to JSON to have a more terminal-friendly output. The top of file ``images.py`` should look like this:
+
+.. code:: python
+
+    import io
+    import os
+    import re
+    import uuid
+    import mimetypes
+
+    import falcon
+    import json
+
+
+    class Collection:
+
+        def __init__(self, image_store):
+            self._image_store = image_store
+
+        def on_get(self, req, resp):
+            # TODO: Modify this to return a list of href's based on
+            # what images are actually available.
+            doc = {
+                'images': [
+                    {
+                        'href': '/images/1eaf6ef1-7f2d-4ecc-a8d5-6e8adba7cc0e.png'
+                    }
+                ]
+            }
+
+            resp.text = json.dumps(doc, ensure_ascii=False)
+            resp.status = falcon.HTTP_200
+
+    def on_post(self, req, resp):
+        name = self._image_store.save(req.stream, req.content_type)
+        resp.status = falcon.HTTP_201
+        resp.location = '/images/' + name
+
+
+Now try the following:
+
+.. code:: bash
+
+    http localhost:8000/images
+
+In response you should get the following data that we statically have put in the code.
+
+.. code::
+
+    {
+        "images": [
+            {
+                "href": "/images/1eaf6ef1-7f2d-4ecc-a8d5-6e8adba7cc0e.png"
+            }
+        ]
+    }
+
+Let's go back to the ``on_get`` method and create a dynamic response. We can use query strings to set maximum image size and get the list of all images smaller than the specified value. We will use method ``get_param_as_int`` to set a default value of ``-1`` in case no ``maxsize`` query string was provided and also to enable a minimum value validation.
+
+.. code:: python
```

Comment:
Don't we need to update any example files on disk? And write tests for them too?

(See also: #2247.)

Reply:
Sure, I'll look into this.

Reply:
@vytas7, I see that a sample usage of query params is being used in the example file for wsgi.
https://github.com/falconry/falcon/blob/65aa7f64d9900380d2246a34d5747782d85d5c8c/examples/things_advanced.py#L158
Are we looking for a more descriptive example?

Reply:
Hi again, no, sorry I was not very clear!

No, it has nothing to do with `things_advanced.py`; what I meant was that we are tracking tutorial files in the Git tree as well, we even have some rudimentary tests for them. In this case it looks like you might need to update [`images.py`](https://github.com/falconry/falcon/blob/master/examples/look/look/images.py) with your additions, and potentially even revise how the file develops further in this tutorial.